### PR TITLE
Fix residual issues with future_array_shapes affecting noise calculation

### PR DESCRIPTION
**Pull request by jsdillon**
This PR replaces a couple of lingering `freq_array[0]` with np.squeeze() which works for both past and future array shapes.

I've added a unit test that fails on the current main branch and passes (I hope!) on this branch. 

Finally, while searching the repo for `freq_array[0]` I found a couple of calculations of delta nu that aren't quite right (they take the middle freq of the first and last bin and divide by the number of bins, when really want you want to do is a diff... that calculation will be wrong by a factor of 2 in the limit where there are only 2 frequency channels). So I fixed those too.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,4 +16,5 @@ hera_pspec/_version.py
 ex.hdf5
 out.h5
 .coverage
-coverage.xml
\ No newline at end of file
+coverage.xml
+.DS_Store
```
Comment (adeliegorce) [inline .gitignore:20]: Is there a reason this changed? it seems unrelated to the PR

Comment (jsdillon) [reply .gitignore:20]: Just added some stuff to avoid Mac's accidentally committing files we don't want.

**Review by adeliegorce — APPROVED** [review]
All good! All tests pass. Thanks!
